File: Celestial-Almanac/src/main/cpp/MathUtils.cpp
// #include <Arduino.h>
#include <stdio.h>
#include <math.h>

#include "MathUtils.h"

MathUtils::MathUtils() {}

double MathUtils::toRadians(double deg) {
  return deg * PI / 180;
//  return (deg * 71) / 4068;
}

double MathUtils::toDegrees(double rad) {
  return rad * 180 / PI;
}

// Sine of angles in degrees
double MathUtils::sind(double x) {
  return sin(toRadians(x));
}

// Cosine of angles in degrees
double MathUtils::cosd(double x) {
  return cos(toRadians(x));
}

// Tangent of angles in degrees
double MathUtils::tand(double x) {
  return tan(toRadians(x));
}

// Normalize large angles
// Degrees
double MathUtils::norm360Deg(double x) {
  while (x < 0) {
    x += 360;
  }
  while (x > 360) {
    x -= 360;
  }
  return x;
}

// Radians
double MathUtils::norm2PiRad(double x) {
  while (x < 0) {
    x += (2 * PI);
  }
  while (x > (2 * PI)) {
    x -= (2 * PI);
  }
  return x;
}

// Cosine of normalized angle (in radians)
double MathUtils::cost(double x) {
  return cos(norm2PiRad(x));
}

File: astro-computer/AstroComputer/src/main/go/Celest/astro/mathutils.go
package astro

import "math"

func ToRadians(deg float64) float64 {
	return deg * math.Pi / 180
}

func ToDegrees(rad float64) float64 {
	return rad * 180 / math.Pi
}

// Sine of angles in degrees
func SinD(x float64) float64 {
	return math.Sin(ToRadians(x))
}

// Cosine of angles in degrees
func CosD(x float64) float64 {
	return math.Cos(ToRadians(x))
}

// Tangent of angles in degrees
func TanD(x float64) float64 {
	return math.Tan(ToRadians(x))
}

// Normalize large angles
// Degrees
func Norm360Deg(x float64) float64 {
	for ok := true; ok; ok = x < 0 {
		x += 360
	}
	for ok := true; ok; ok = x > 360 {
		x -= 360
	}
	return x
}

// Radians
func Norm2PiRad(x float64) float64 {
	for ok := true; ok; ok = x < 0 {
	    x += (2 * math.Pi)
    }
	for ok := true; ok; ok = x > (2 * math.Pi) {
		x -= (2 * math.Pi)
 	}
	return x
}

// Cosine of normalized angle (in radians)
func CosT(x float64) float64 {
  return math.Cos(Norm2PiRad(x))
}
